### PR TITLE
Add perf test instructions

### PR DESCRIPTION
**Pull request by bojanserafimov**
We should all know these things by now but IMO it's useful for onboarding.

Also should help people outside Neon, who might run a debug build and wonder why it's slow

Resolves https://github.com/neondatabase/neon/issues/2577

### CODE REVIEW

**Review by hlinnaka — APPROVED** [review]

File: test_runner/performance/README.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+# Running locally
+
+First make a release build. The profiling flag is optional, used only for tests that
+generate flamge graphs. The `-s` flag just silences a lot of output, and makes it
+easier to see if you have compile errors without scrolling up.
+`BUILD_TYPE=release CARGO_BUILD_FLAGS="--features=testing,profiling" make -s -j8`
```
Comment (alexstanovoy) [inline test_runner/performance/README.md:6]: This doesn't work in my case. Even `cargo build --features profiling` with the following error:
https://pastebin.com/d1GWtSKx
Haven't investigated that, will try in the morning.

**Review by alexstanovoy — CHANGES_REQUESTED** [review]

File: test_runner/performance/README.md (hunk 1 above)
Comment (bojanserafimov) [reply test_runner/performance/README.md:6]: Looks like the libc crate exports different types on linux (`timer_t`) vs osx (`time_t`), and we use `timer_t` in pprof https://github.com/neondatabase/pprof-rs/blob/wallclock-profiling/src/timer.rs#L24

Comment (bojanserafimov) [reply test_runner/performance/README.md:6]: I added a note explaining this.

File: test_runner/performance/README.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
+# Running locally
+
+First make a release build. The profiling flag is optional, used only for tests that
+generate flamge graphs. The `-s` flag just silences a lot of output, and makes it
```
Comment (LizardWizzard) [inline test_runner/performance/README.md:4]: typo
```suggestion
generate flame graphs. The `-s` flag just silences a lot of output, and makes it
```

**Pull request merged**